accepted/wasmtime-one-dot-oh.md

```diff
@@ -0,0 +1,368 @@
+proposals in flight to extend and modify WebAssembly. The nature of Wasmtime and
+its goals mean that **Wasmtime's stability story will not include API stability
+in the literal sense of "Wasmtime will always be 1.0 and never 2.0"**.
+
+For Rust users this may feel a bit discombobulating where Rust is likely to
+always be "1.x", but this RFC is proposing that Wasmtime will likely never
+remain indefinitely "1.x". Instead users and developers can expect that Wasmtime
+will regularly release new major versions of its tooling. With the expectation
+that Wasmtime will release new major versions it enables developers of Wasmtime
+itself to land API-breaking changes. This makes it much easier to implement new
+WebAssembly proposals, improve the runtime, fix bugs, make APIs more ergonomic,
+etc. These are seen as critical abilities needed to improve Wasmtime over time.
+As mentioned in the motivation section, Wasmtime developers themselves should
+not feel overly burdened or having to contort their code to land changes which
+are seen as important. Allowing breaking changes is seen as key to enabling
+this.
+
+Naturally, though, the world is not entirely made up of Wasmtime developers.
+Instead it's much more likely that the number of users of Wasmtime far
+outweighs the number of developers on Wasmtime. A stream of constant breaking
+changes makes it feel harder to use Wasmtime since it can feel more difficult to
+keep up with the latest and receive bug fixes and new features. Wasmtime's
+stability story, consequently, will not be on the end of the spectrum where
+breaking changes are landed willy-nilly. This means that there's a tension set
+up between developers of Wasmtime and users of Wasmtime, but this is to be
+expected from the stable release of a project.
+
+On a spectrum of "Wasmtime never breaks any API ever" to "Wasmtime breaks all
+its APIs every release" Wasmtime will reside towards the first half here. In
+other words Wasmtime developers will be expected to rarely and deliberately
+break existing APIs, but will also be allowed to break APIs as necessary. The
+major goal here is "hassle-free upgrades" inspired by the Rust release process
+where API stability in the absolute literal sense is deterimental to the
+development of the project, but deliberate effort and work is done to ensure
+upgrades are as seamless as possible.
+
+Users should expect that they will regularly need to increase the major version
+of Wasmtime to upgrade Wasmtime usage in their projects. Furthermore users
+should expect that between upgrades they may need to tweak some APIs and update
+them to continue to use the latest-and-greatest Wasmtime. At the same time
+though users should expect that they will not be blindsided by major
+refactorings. For example changes like [Wasmtime's new API][new-api] will still
+go through the RFC process and will be deliberately scheduled and widely
+advertised before they're released.
```

Comment:
Maybe worth explicitly mentioning that, even though they will go through the RFC process when we need them, we expect such large changes to be rare.